Signal/src/ViewControllers/OWSTableViewController.h

```diff
@@ -32,15 +32,30 @@ NS_ASSUME_NONNULL_BEGIN
 #pragma mark -
 
 typedef NS_ENUM(NSInteger, OWSTableItemType) {
+    OWSTableItemTypeDefault,
     OWSTableItemTypeAction,
 };
 
 typedef void (^OWSTableActionBlock)();
+typedef UITableViewCell * (^OWSTableCustomCellBlock)();
 
 @interface OWSTableItem : NSObject
 
-+ (OWSTableItem *)actionWithTitle:(NSString *)title
-                      actionBlock:(OWSTableActionBlock)actionBlock;
++ (OWSTableItem *)itemWithTitle:(NSString *)title actionBlock:(OWSTableActionBlock)actionBlock;
+
++ (OWSTableItem *)itemWithCustomCell:(UITableViewCell *)customCell
+                     customRowHeight:(CGFloat)customRowHeight
+                         actionBlock:(OWSTableActionBlock)actionBlock;
+
++ (OWSTableItem *)itemWithCustomCellBlock:(OWSTableCustomCellBlock)customCellBlock
+                          customRowHeight:(CGFloat)customRowHeight
+                              actionBlock:(OWSTableActionBlock)actionBlock;
+
++ (OWSTableItem *)itemWithCustomCellBlock:(OWSTableCustomCellBlock)customCellBlock
+                              actionBlock:(OWSTableActionBlock)actionBlock;
+
+- (UITableViewCell *)customCell;
+- (NSNumber *)customRowHeight;
```

Comment:
I haven't gotten there yet, but I'd eventually like to let these items specify "custom row height" using either a fixed value or a block.  Once we add block support, this will be a legit method, not a getter.